lib/commands/audit.js

```diff
@@ -176,35 +176,63 @@ class VerifySignatures {
     })
   }
 
-  async getVerifiedInfo (edge) {
+  getEdgeType (edge) {
+    return edge.optional ? 'optionalDependencies'
+      : edge.peer ? 'peerDependencies'
+      : edge.dev ? 'devDependencies'
+      : 'dependencies'
+  }
+
+  getEdgeSpec (edge) {
     let alias = false
     try {
       alias = npa(edge.spec).subSpec
     } catch (err) {
     }
-    const spec = npa(alias ? alias.name : edge.name)
-    const node = edge.to || edge
-    const { location } = node
-    const { version } = node.package || {}
-
-    const type = edge.optional ? 'optionalDependencies'
-      : edge.peer ? 'peerDependencies'
-      : edge.dev ? 'devDependencies'
-      : 'dependencies'
-
-    // Skip local workspaces
-    if (node.isWorkspace) {
+    let spec
+    try {
+      spec = npa(`${alias ? alias.name : edge.name}@${edge.spec}`)
+    } catch (_) {
+      // Skip packages with invalid spec
       return
     }
+    return spec
+  }
+
+  buildRegistryConfig (registry) {
+    const keys = this.keys.get(registry) || []
+    const registryConfig = {}
+    const parsedRegistry = new URL(registry)
+    const regKey = `//${parsedRegistry.host}${parsedRegistry.pathname}`
+    registryConfig[`${regKey}:_keys`] = keys
+    return registryConfig
```

Comment:
```suggestion
    const parsedRegistry = new URL(registry)
    const regKey = `//${parsedRegistry.host}${parsedRegistry.pathname}`
    return {
      `${regKey}:_keys`: keys
    }
```
it's better practice and more performant to avoid mutating objects :-)

Reply:
nice one, thanks!